### ISSUE TITLE
odin/telnet/stream.cpp does not compile with Clang

### ISSUE THREAD

**Issue by GoogleCodeExporter** [KazDragon/paradice9#141]
```
Error message:

In file included from odin/telnet/stream.cpp:28:
./odin/telnet/detail/generator.hpp:82:65: error: implicit instantiation of
      undefined template
      'odin::telnet::detail::generator_grammar<std::back_insert_iterator<std::vector<unsigned
      char, std::allocator<unsigned char> > > >'
        static generator_grammar<BOOST_TYPEOF(output_iterator)> g;
                                                                ^
./odin/telnet/detail/generator.hpp:45:7: note: template is declared here
class generator_grammar;
      ^
1 error generated.

-----

In short, there is a mutual dependency between the grammar and its output 
iterator type that just happens to compile with MSVC and g++.

```

Original issue reported on code.google.com by `matthew....@gmail.com` on 12 Dec 2013 at 2:22
